lib/pg_query/parse.rb

```diff
@@ -3,7 +3,13 @@ def self.parse(query)
     result, stderr = parse_protobuf(query)
 
     begin
-      result = PgQuery::ParseResult.decode(result)
+      result = if PgQuery::ParseResult.method(:decode).arity == 1
+                 PgQuery::ParseResult.decode(result)
+               elsif PgQuery::ParseResult.method(:decode).arity == -1
+                 PgQuery::ParseResult.decode(result, max_recursion_depth: 100)
```

Comment:
Back in v1 the nesting limit was 1000, would you be willing to restore it to 1000 again? That would be comfortably enough for my use case.

Reply:
Good question - I don't see a problem with that, but it'd be good to test if there are any other constraints (e.g. stack size).

Do you have an example query that is deeper than 100 levels? (most that I could find easily were < 100)

Reply:
Good idea. Here is a query that is deeper than 100 levels. Join conditions get nested within each other in the tree, so a query with enough joins will do the trick. I haven't measured its depth but I think this one is at least 200 levels deep.

```ruby
'SELECT * FROM foo ' + (1..100).to_a.map { |i| "JOIN foo_#{i} ON foo.id = foo_#{i}.foo_id" }.join(' ')
```

Reply:
The good news is, the fix for this has been merged in Protobuf v3.20.0, so we can now rely on the recursion depth setting when the newer release is active. PR updated, and I'll merge that shortly.